### DISCUSSION TITLE
차원 축소

### DISCUSSION THREAD

**Discussion by K-yng** [kangjun205/team_F4#16] [General]
# 서론
- 차원 축소 모델들에 대해 정리하는 도중에, 전에 재겸님이 PCA와 Auto-Encoder의 차이가 선형, 비선형 관계로 알고 있다고 말씀하신 게 생각이 나서 관련된 내용을 정리해봤습니다.
- 먼저 각종 차원 축소 모델들의 개념에 대해 간단히 정리해보고, 마지막에 PCA와 Auto-Encoder의 차이에 대해서 정리해보겠습니다.
# 차원 축소 모델
## PCA
- 변동성이 가장 큰 축이 가장 많은 정보를 담고 있을 것이라는 믿음을 기반으로 함
- 변동성이 가장 큰 축을 고차원 데이터 $X$에 대한 선형 변환의 변동을 최대화 하는 방식으로 찾으려 함
## t-SNE
- 고차원 데이터 샘플 $(x_1, x_2, \dots, x_n)$로부터, 2 or 3차원의 데이터 $(y_1, y_2, \dots, y_n)$을 도출하려 함
- 고차원 데이터 간의 유사도를 가장 잘 근사하는 저차원 데이터를 추론하는 것이 목표
- 고차원 데이터에 대해서는 gaussian kernel 기반의 유사도 계산 및 distribution 정의
- 저차원 데이터에 대해서는 t-distribution 기반의 유사도 계산 및 distribution 정의
- 두 분포간의 KL-divergence를 최소화하는 방향으로 학습
## Isomap
- MDS기반의 비선형 차원 축소 모델
- MDS는 고차원 데이터 간의 Euclidean distance를 가장 잘 근사하는 저차원 데이터를 추론하는 것을 목적으로 하는 모델
- MDS는 고차원 데이터 간의 Euclidean distance를 활용했지만, Isomap에서는 graph 기반의 활용
- 각각의 데이터 포인트에 대해서 가장 가까이 있는 k개의 데이터 포인트를 구해서 연결함
- 연결된 거리는 Euclidean distance로 정의
- 이렇게 연결된 그래프를 기반으로, 각 포인트간의 거리를 그래프 상의 최단 거리로 설정
  - ex) A $\rightarrow$ B : 5, B $\rightarrow$ C : 3 $\Rightarrow$ A $\rightarrow$ C : 8
## Auto-Encoder
- 고차원 데이터를 저차원 데이터로 변환한 뒤 다시 원래대로 변환하는 과정을 담은 신경망 기반 모델
- 고차원에서 저차원으로 변환하는 과정을 encoder라 하고, 저차원에서 고차원으로 변환하는 과정을 decoder라 함
- 고차원 데이터 $x_i$에 대해서 인코더와 디코더를 적용한 결과 $D(E(x_i))$와의 차이가 최소가 되는 방향으로 모델 학습
- 인코더를 통과한 결과를 차원 축소된 데이터로 활용
# PCA VS Auto-Encoder
- PCA는 데이터에 대한 선형 변환을 통해, 변동이 가장 커지는 축을 찾으려 함
- 반면에, Auto-Encoder는 신경망 기반의 모델이기 때문에 데이터에 대한 비선형적인 변환이 일어남

**Comment by sunvvis** [comment]
개추..드림니더...😂👍👍👍
